scalapb-runtime/src/main/scala/scalapb/CodedOutputStreamEncoder.scala

```diff
@@ -0,0 +1,87 @@
+package scalapb
+
+import com.google.protobuf.{CodedInputStream, CodedOutputStream, InvalidProtocolBufferException}
+
+import scala.language.implicitConversions
+
+trait CodedOutputStreamEncoder[T] {
+  def serializedSizeNoTag(value: T): Int
+  def writeTo(output: CodedOutputStream, value: T): Unit
+}
+
+case object CodedOutputStreamEncoder {
+  sealed trait Predef
+  implicit case object StringCodedOutputStreamEncoder
+      extends CodedOutputStreamEncoder[String]
+      with Predef {
+    override def serializedSizeNoTag(value: String): Int =
+      CodedOutputStream.computeStringSizeNoTag(value)
+
+    override def writeTo(output: CodedOutputStream, value: String): Unit =
+      output.writeStringNoTag(value)
+  }
+
+  implicit def messageCodedOutputStreamEncoder[M <: GeneratedMessage]: CodedOutputStreamEncoder[M] = new CodedOutputStreamEncoder[M] with Predef {
+    override def serializedSizeNoTag(value: M): Int = value.serializedSize
+
+    override def writeTo(output: CodedOutputStream, value: M): Unit = value.writeTo(output)
+  }
+}
+
+trait CodedOutputStreamDecoder[T] {
```

Comment:
File name suggested only the Encoder would be here. Maybe split to two files or rename the file to something like `CodedOutputStreamFormatters`...